pos_order_reorder/static/src/js/Screens/TicketScreen/ControlButtons/ReorderButton.js

```diff
@@ -0,0 +1,74 @@
+odoo.define("pos_order_reorder.ReorderButton", function (require) {
+    "use strict";
+
+    const {useListener} = require("@web/core/utils/hooks");
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+    const {Orderline} = require("point_of_sale.models");
+
+    class ReorderButton extends PosComponent {
+        setup() {
+            super.setup();
+            useListener("click", this._onClick);
```

Comment:
Is the `useListener` intentional here? to be able to have contextual / overriden button templates that don't define the `t-on-click` directive (abstract behavior, without template)?

If that's not the case, since `useListener` is a **deprecated** hook, please prefer to bind the click event on the template directly:

```xml
<t t-name="ReorderButton" owl="1">
        <span class="control-button" t-on-click="_onClick">
            <i class="fa fa-refresh" />
            <span></span>
            <span>Re-order</span>
        </span>
    </t>
```
With that you can completely remove that setup() method.